src/Esfa.Vacancy.UnitTests/CreateApprenticeship/Application/GivenACreateApprenticeshipRequestValidator/WhenValidatingTrainingCode/AndTrainingTypeIsFramework.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Esfa.Vacancy.Application.Commands.CreateApprenticeship;
+using Esfa.Vacancy.Application.Commands.CreateApprenticeship.Validators;
+using Esfa.Vacancy.Domain.Entities;
+using Esfa.Vacancy.Domain.Validation;
+using FluentValidation.TestHelper;
+using NUnit.Framework;
+using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.AutoMoq;
+
+namespace Esfa.Vacancy.UnitTests.CreateApprenticeship.Application.GivenACreateApprenticeshipRequestValidator.WhenValidatingTrainingCode
+{
+    [TestFixture]
+    public class AndTrainingTypeIsFramework
+    {
+        [TestCase(
+            null,
+            "'Training Code' should not be empty.",
+            true,
+            TestName = "And is null then is invalid.")]
+        [TestCase(
+            "",
+            "'Training Code' should not be empty.",
+            true,
+            TestName = "And is empty then is invalid.")]
+        [TestCase(
+            "123",
+            "'Training Code' should be a in format '###-##-#' when Training Type is Framework.",
+            true,
```

Comment:
As per another comment - regex will take 9999-99-99 as valid - is the error message consistent with this?

Reply:
fixed